wicket-core/src/main/java/org/apache/wicket/core/request/handler/PageProvider.java

```diff
@@ -54,15 +52,14 @@
 
 	private final Integer pageId;
 
-	private IPageSource pageSource;
-
-	private transient IRequestablePage pageInstance;
-	private boolean pageInstanceIsFresh;
+	private transient IPageSource pageSource;
```

Comment:
It can lead to inconsistent behaviour indeed, but not NPE since it will default to the application one. I will support the removal of IClusterable interface. From what I looked in the code, page provider is just a core's internal communicator of pages construction and fetch logic. It's used during the work to handle a page request and never shared on the session. I would bet that at some point a developer needed ou wanted a page provider as a page attribute and ended up marking it as serializable.